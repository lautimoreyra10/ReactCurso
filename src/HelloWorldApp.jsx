// function App () {
//     //document.createElement...
//     return (<h1>Hola Mundo</h1>);
// }


// export default App;

// Esta es otra forma de hacerlo, utilizando el shortcut "rafc"
export const HelloWorldApp = () => {
    
  return (
    <h1>Hello World App</h1>
  );
}

//Otra forma mas
//export const HelloWorldApp = () => (<h1>Arrow Function</h1>)
