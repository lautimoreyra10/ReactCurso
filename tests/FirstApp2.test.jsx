/** @jest-environment jsdom */
import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


describe('Pruebas en FirstApp', () => {

    const title = 'Hola, soy yo'
    const subTitle = 'Hola soy un subtitulo'

    test('Debe hacer match con snapshot', () => {

        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();

    });
    test('Debe mostrar el mensaje "Hola, soy yo"', () => {
        render(<FirstApp title={title} />);
        // screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
    });
    test('debe mostrar el tÃ­tulo en un h1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });
    test('debe mostrar el subtitulo enviado por props', () => {
        render(
            <FirstApp
                title={title}
                subTitle={subTitle}
            />
        );
        expect(screen.getAllByText(subTitle).length).toBe(2);
    });
});