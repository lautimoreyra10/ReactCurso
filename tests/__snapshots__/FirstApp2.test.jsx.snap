// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer match con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy yo
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Lautaro Moreyra
  </p>
</div>
`;
